jujugui/static/gui/src/test/test_ghost_deployer_extension.js

```diff
@@ -157,6 +157,18 @@ describe('Ghost Deployer Extension', function() {
     assert.equal(services.item(3).get('name'), 'mysql-a');
   });
 
+  it('increments the name with deleted services', function() {
+    var charm = makeCharm();
+    ghostDeployer.db.services = new Y.juju.models.ServiceList();
+    var services = ghostDeployer.db.services;
+    services.ghostService(charm);
+    services.ghostService(charm);
+    services.item(0).destroy();
```

Comment:
Could you also add a test where you remove a middle service and make sure it still increments the last one so they don't conflict.